src/main/scala/com/ing/wbaa/airlock/proxy/api/directive/ProxyDirectives.scala

```diff
@@ -62,9 +62,25 @@ object ProxyDirectives extends LazyLogging {
         case Tuple1(optionalSessionToken) =>
           headerValue[AwsAccessKey](extractAuthorizationS3) tmap { case Tuple1(awsAccessKey) =>
 
+            // aws is passing subdir in prefix parameter if no object is used, eg. list bucket objects
+            val s3path = httpRequest.uri.rawQueryString match {
+              case Some(queryString) if queryString.contains("prefix") =>
+                val queryPrefixPair = queryString
+                  .split("&")
+                  .filter(_.contains("prefix"))
+                  .head.split("=")
+
+                if (queryPrefixPair.length == 2) {
+                  Uri.Path(s"${httpRequest.uri.path}/${queryPrefixPair.last.replace("%2F", "/")}")
```

Comment:
I noticed there is a field in the URL we get from AWS when listing subfolders (`http://127.0.0.1:8010/demobucket?prefix=test%2Fsub&delimiter=%2F&encoding-type=url`) that specifies the delimiter used (`%2F`).
Maybe we can use that delimiter to do the replace here